### PR TITLE
[Dependabot] Ignore pulumi dependencies coming from test-infra

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

[Dependabot] Ignore dependencies coming from test-infra

### Motivation

https://datadoghq.atlassian.net/browse/ADXT-375

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -146,6 +146,20 @@ updates:
...
+      - dependency-name: github.com/alessio/shellescape
+      - dependency-name: github.com/cenkalti/backoff/v4
+      - dependency-name: github.com/google/gofuzz
+      - dependency-name: github.com/aws/*
+      - dependency-name: github.com/pulumi/*
+      - dependency-name: github.com/pulumiverse/pulumi-time/sdk
+      - dependency-name: github.com/samber/lo
+      - dependency-name: github.com/stretchr/testify
+      - dependency-name: golang.org/x/exp
+      - dependency-name: gopkg.in/yaml.v3
```
Comment (pducolin) [inline .github/dependabot.yaml:161]: **💭 thought**
Wondering if we should limit to dependencies that cause plugins downloads, so `pulumi*` ones. Other dependencies are safe to bump, even if we should keep them in sync

Comment (FlorentClarret) [reply .github/dependabot.yaml:161]: That's a good point. Let me change this!

**Review by amenasria — APPROVED** [review]
LGTM !

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -146,6 +146,10 @@ updates:
       # requires to also bump `TEST_INFRA_DEFINITIONS_BUILDIMAGES` inside `.gitlab/common/test_infra_version.yml`
       # and dependabot isn’t able to keep those two synchronized.
       - dependency-name: github.com/DataDog/test-infra-definitions
+      # Do not bump dependencies that come from test-infra-definitions, they will be bumped when we'll bump the version
+      # https://datadoghq.atlassian.net/browse/ADXT-375
+      - dependency-name: github.com/pulumi/*
```
Comment (pducolin) [inline .github/dependabot.yaml:151]: To include [pulumiverse/pulumi-time](https://github.com/DataDog/datadog-agent/pull/27440/files#diff-fee3041315eef2d6446d1e0f565e42851503923fb40e53cf4c91ac5641ebc4e0R185) 
```suggestion
      - dependency-name: github.com/pulumi*
```

**Review by pducolin — APPROVED** [review]

**Pull request merged**